spring-integration-mail/src/main/java/org/springframework/integration/mail/ImapMailReceiver.java

```diff
@@ -250,16 +253,13 @@ protected void setAdditionalFlags(Message message) throws MessagingException {
 
 		@Override
 		public void run() {
-			try {
-				Folder folder = getFolder();
-				logger.debug("Canceling IDLE");
-				if (folder != null) {
-					folder.isOpen(); // resets idle state
-				}
-			}
-			catch (Exception ignore) {
+			Folder folder = getFolder();
+			logger.debug("Canceling IDLE");
+			if (folder != null) {
+				folder.isOpen(); // resets idle state
 			}
 		}
```

Comment:
I presume this was to avoid log noise. Why change? If your intention is to alert the user of the problem, it would be better to log rather than the task executor printing the stack trace.